EIPS/eip-3030.md

```diff
@@ -0,0 +1,397 @@
+
+## Specification
+
+### `GET /upcheck`
+
+_**Responses**_
+
+Success | <br>
+--- | ---
+Code | `200`
+Content | `{"status": "OK"}`
+
+---
+
+### `GET /publicKeys`
+
+_**Responses**_
+
+Success | <br>
+--- | ---
+Code | `200`
+Content | `{"public_keys": "[public_key_hex_string_without_0x]"}`
+
+---
+
+### `POST /sign/:public_key`
+
+URL Parameter | <br>
+--- | ---
+`:public_key` | `public_key_hex_string_without_0x`
+
+_**Request**_
+
+JSON Body | <br> | <br>
+--- | --- | ---
+`signingRoot` | **Required** | A string representation of the hexadecimal value to be signed
+<br> | Optional | Any other field will be ignored by the signer
+
+_**Responses**_
+
+Success | <br>
+--- | ---
+Code |  `200`
+Content | `{"signature": "<signature_hex_string>"}`
+
+_or_
+
+Error | <br>
+--- | ---
+Code |  `400`
+Content | `{"error": "<Bad Request Error Message>"}`
+
+_or_
+
+Error | <br>
+--- | ---
+Code |  `404`
+Content | `{"error": "Key not found: <public_key_hex_string_without_0x>"}`
+
+---
+
+## Rationale
+
+### URL parameter `:public-key` without `0x`
+
+By relaxing the constraint of requiring hexadecimal input, clients can opt-in to using custom ids for their private keys, and perform the adequate retrieving and matching with the secret key within the signer. That is, if the remote signer implementer allows for this feature.
+
+### UNIX philosophy: Simple API
+
+This API specification contains only three methods: one for **status**, one for **listing the available keys**, and one to **produce a signature**. There are no methods for authentication, pre-image validation, key management, nor transport encryption.
+
+The following subsections discuss aspects to be considered by the client implementers relative to these subjects.
+
+#### Implementation of additional features
+
+From an API pipeline view, we have two nodes: The validator client (1) that makes requests to the remote signer (2). A more sophisticated chain can be built by introducing elements between these two nodes. Either by setting up reverse proxy services, or by adding plugins to the remote signer implementation.
+
+#### Authentication
+
+Can be accomplished through the use of an HTTP Request Header. There are several ways to negotiate and issue a valid token to authenticate the communication between the validator client and the remote signer, each of them with potential drawbacks (e.g replay attacks, challenges in distributing the token to the validator client, etc.). In general, any method of authentication must be combined with transport encryption to be effective.
```

Comment:
There is no need for authentication if the requestor is not privileged. And it wouldn't be privileged if you removed the need for the signer to trust the `signingRoot`. 
So in essence, you now force the validator to store some sealer-credential, instead of a sealing-key -- just swapping one set of secrets for another.

Reply:
I agree. This rationale should be more emphatic on the point about swapping of secrets, since it beats the purpose of this API. Please notice that the following paragraph advises the reader to explore the use of network Access Control Lists (ACLs).